app.ts

```diff
@@ -0,0 +1,238 @@
+					}
+					return parseFloat(input as string);
+				case ("number"):
+					return input as number;
+				default:
+					return 0;
+			}
+			throw new Error("Error");
+		}
+		function calculateToId(fromId: number): number {
+			const possibleRecordsData = Math.floor((window.innerHeight - ($("#form-content").innerHeight() as number)) / 37);
+			const possibleId = fromId + possibleRecordsData;
+
+			let recordDisplayset = 0;
+			switch (recordDisplayset) {
+				case 0:
+					window.innerHeight <= 646;
+					break;
+				case 1:
+					window.innerHeight <= 969;
+					break;
+				case 2:
+					window.innerHeight <= 1938;
+					break;
+				default:
+					recordDisplayset = 15;
+					break;
+			}
+			return recordDisplayset + possibleId;
+		}
+		function nextPageResize(previous: number[]): number {
+			const fromID = ConvertNumber(previous.sort((a, b) => { return a - b })[0]);
+			const toID = ConvertNumber(previous.sort((a, b) => { return a - b })[1]);
+			const documentHeight = $(window).innerHeight() as number - ($(`#table-row-${fromID}`).height() as number);
+			for (let i = fromID; i <= toID; i++) {
+				const elementHeightOffset = ($(`#table-row-${i}`).offset() as JQueryCoordinates).top;
+				if (elementHeightOffset < documentHeight) continue;
+				return i;
+			}
+			return toID;
+		}
+		function previousPageResize(previous: number[]): number[] {
+			const toId = calculateToId(previous[0] - (nextPageResize(previous) - previous[0]));
+			return [previous[0] - (nextPageResize(previous) - previous[0]), toId];
+		}
+		window.onload = async () => {
+			try {
+				previous = await LoadPageContent(0, calculateToId(0));
+				$("#previous").click(async () => {
+					const CountData = await getRecordCountCall();
+					previous = previousPageResize(previous);
+					let fromId = previous[0] >= 0 ? previous[0] : 0;
+					const possibleStep = calculateToId(fromId) - fromId;
+					let toId = (previous[0] >= 0 ? previous[1] : possibleStep);
+					fromId = fromId == CountData - 1 ? fromId - possibleStep : fromId;
+					toId = toId <= CountData - 1 ? toId : CountData - 1;
+					previous = await LoadRecordsData(fromId, toId);
+				});
+			} catch (error) {
+			}
+			$("#next").click(async () => {
+				try {
+					const recordCount = await getRecordCountCall();
+					const fromId = nextPageResize(previous);
+					const possibleStep = calculateToId(fromId) - fromId;
+					if (fromId <= recordCount - possibleStep - 1) {
+						const toId = fromId + possibleStep <= recordCount - 1 ? fromId + possibleStep : recordCount - 1;
+						previous = await LoadRecordsData(fromId, toId);
+					} else if (fromId <= recordCount - 1) {
+
+						previous = await LoadRecordsData(recordCount - 1 - (calculateToId(fromId) - fromId), recordCount - 1);
+					} else {
+						console.log("Test is working ");
+					}
+				} catch (error) {
+				}
+			});
+			$("#go-button").click(async () => {
+				try {
+					const recordCount = await getRecordCountCall();
```

Comment:
You are allowed to skip this step. For this project the record count will not change after the initial call.